### PR TITLE
Scorers string

### LINKED ISSUE #1630: Refactor alternative metric mechanism to support new metrics and string metrics in vision
**Is your feature request related to a problem? Please describe.**
There are several basic metrics that are currently not supported and we should implement, those include:
Classification: Sensitivity (recall), [specificity](https://www.jigsawacademy.com/sensitivity-vs-specificity-in-logistic-regression/), FPR, FNR, Precision-recall AUC
Object detection: in other issue - https://github.com/deepchecks/deepchecks/issues/1634

**Describe the solution you'd like**
Implement the metrics internally, make sure those metrics are implemented support the DeepcheckScorer class api. In addition keep in mind those metrics should also be applicable to vision classification tasks.
Add unit tests to make sure those metrics return the correct value + work properly with the DeepcheckScorer API

### PR DESCRIPTION
**Pull request by JKL98ISR**
closes #1630

### CODE REVIEW

File: deepchecks/tabular/metric_utils/metrics.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        scorer name
+    """
+
+    def __init__(self, scorer: t.Union[str, t.Callable], name: str):
+        self.name = name
+        if isinstance(scorer, str):
+            formated_scorer_name = scorer.lower().replace('sensitivity', 'recall').replace(' ', '_')
+            if formated_scorer_name in _func_dict:
+                self.scorer = _func_dict[formated_scorer_name]
+            else:
```
Comment (Nadav-Barak) [inline deepchecks/tabular/metric_utils/metrics.py:101]: would add code comment - this uses sklearn string to scorer mechanism https://scikit-learn.org/stable/modules/model_evaluation.html

File: deepchecks/tabular/metric_utils/metrics.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    name : str
+        scorer name
+    """
+
+    def __init__(self, scorer: t.Union[str, t.Callable], name: str):
+        self.name = name
+        if isinstance(scorer, str):
+            formated_scorer_name = scorer.lower().replace('sensitivity', 'recall').replace(' ', '_')
+            if formated_scorer_name in _func_dict:
+                self.scorer = _func_dict[formated_scorer_name]
```
Comment (Nadav-Barak) [inline deepchecks/tabular/metric_utils/metrics.py:100]: in those case,  the name displayed in graph different (without the per_class)

Comment (Nadav-Barak) [reply deepchecks/tabular/metric_utils/metrics.py:100]: i.e change self.name

File: deepchecks/tabular/metric_utils/metrics.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+}
+
+DEFAULT_SCORERS_DICT = {
+    TaskType.BINARY: DEFAULT_BINARY_SCORERS,
+    TaskType.MULTICLASS: DEFAULT_MULTICLASS_SCORERS,
+    TaskType.REGRESSION: DEFAULT_REGRESSION_SCORERS
+}
+
+
+_func_dict = {
```
Comment (Nadav-Barak) [inline deepchecks/tabular/metric_utils/metrics.py:73]: Additional thing needed here - all the regression stuff + 'Accuracy'

File: deepchecks/tabular/metric_utils/metrics.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+    Parameters
+    ----------
+    scorer : t.Union[str, t.Callable]
+        sklearn scorer name or callable
+    name : str
+        scorer name
+    """
+
+    def __init__(self, scorer: t.Union[str, t.Callable], name: str):
```
Comment (Nadav-Barak) [inline deepchecks/tabular/metric_utils/metrics.py:95]: Additional request - add a class field indicating whether the scorer returns a float or a list of float + a public method is_scalar_scorer

Comment (JKL98ISR) [reply deepchecks/tabular/metric_utils/metrics.py:95]: not now

File: deepchecks/tabular/metric_utils/metrics.py (hunk 2 above)
Comment (JKL98ISR) [reply deepchecks/tabular/metric_utils/metrics.py:100]: yes

File: deepchecks/tabular/metric_utils/metrics.py (hunk 1 above)
Comment (JKL98ISR) [reply deepchecks/tabular/metric_utils/metrics.py:101]: this is already there, need to add to vision

**Review by Nadav-Barak — APPROVED** [review]
Add test to see at least one of the checks works with getting a list of str to alternative_scorers
+ see what happens if you pass str that do not exist in sklearn and in ours

File: deepchecks/vision/metrics_utils/metrics.py [hunk 5]
```diff
@@ -45,18 +49,58 @@ def get_default_object_detection_scorers() -> t.Dict[str, Metric]:
     }
 
 
+_func_naming_dict = {
+    'precision': 'Precision',
+    'recall': 'Recall',
```
Comment (noamzbr) [inline deepchecks/vision/metrics_utils/metrics.py:54]: I think this are the per class options, right? If so, perhaps for consistency we should call these precision_per_class etc
@Nadav-Barak WDYT?

Comment (JKL98ISR) [reply deepchecks/vision/metrics_utils/metrics.py:54]: doesn't matter much tbh
default behavior will remain the same
don't mind changing them but not sure if it will make much sense cuz all the name will be longer and all the threshold metrics names won't correspond the the name in that dict

File: deepchecks/vision/metrics_utils/metrics.py [hunk 6]
```diff
@@ -45,18 +49,58 @@ def get_default_object_detection_scorers() -> t.Dict[str, Metric]:
...
+    'sensitivity': 'Recall',
+    'average_precision': 'Average Precision',
+    'average_recall': 'Average Recall',
```
Comment (Nadav-Barak) [inline deepchecks/vision/metrics_utils/metrics.py:57]: If not too messy, also add mean_average_precision and mean_average_recall options

Comment (JKL98ISR) [reply deepchecks/vision/metrics_utils/metrics.py:57]: kinda is

**Review by yromanyshyn — APPROVED** [review]
